components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -357,24 +358,36 @@ export class BitbucketServerApi {
         if (isCancelled()) {
             return [];
         }
-        const cap = (query?.cap || 0) > 0 ? query.cap! : 10;
-        let requestsLeft = cap;
-        const limit = `limit=${(query?.limit || 0) > 0 ? query.limit! : 1000}&`;
-        const permission = query.permission ? `permission=${query.permission}&` : "";
-        const runQuery = async (params: string) => {
+
+        const fetchRepos = async (params: { [key: string]: string } = {}) => {
```

Comment:
I noticed we weren't url encoding values correctly here, so I refactored parameters here to use `URLSearchParams` instead of string concatenating.